contrib/node/src/python/pants/contrib/node/subsystems/resolvers/node_resolver_base.py

```diff
@@ -35,3 +36,25 @@ def _copy_sources(self, target, results_dir):
       dest = os.path.join(results_dir, os.path.relpath(source, source_relative_to))
       safe_mkdir(os.path.dirname(dest))
       shutil.copyfile(os.path.join(buildroot, source), dest)
+
+  def _get_target_from_package_name(self, target, package_name, file_path):
+    """Get a target from its dependency tree given a package name and relative file path.
```

Comment:
This is not going to operate on the dependency tree: if I'm understanding it correctly, it will find the dep with the matching package name in the direct dependencies of the given target.